File: client/lib/services/tag_service.dart
import 'package:client/datas/models/tag.dart';
import 'package:client/datas/responses/tags_response.dart';
import 'package:client/http/http.dart';

final _http = Http();

class TagService {
  Future<List<Tag>> getTags() async {
    var res = await _http.get("tag");
    TagsResponse tagsResponse = TagsResponse.fromJson(res.response);
    return tagsResponse.tags;
  }
}

File: client/src/app/core/services/tag.service.ts
import { environment } from '../../../environments/environment';
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { TagsResponse } from '@shared/responses/tags.response';
import { Tag } from '@shared/models/tag.model';
import { map } from 'rxjs/operators';

@Injectable({
    providedIn: 'root'
})
export class TagService {

    constructor(private http: HttpClient) { }

    getTags(): Observable<Tag[]> {
        return this.http.get<TagsResponse>(environment.apiUrl + 'tag')
            .pipe(
                map((response: TagsResponse) => {
                  return response.data;
                })
              );
    }

}